contracts/modules/commons/ModuleCalls.sol

```diff
@@ -68,23 +103,25 @@ abstract contract ModuleCalls is IModuleAuth {
 
   /**
    * @notice Verify if a nonce is valid
-   * @param _nonce Nonce to validate
+   * @param _rawNonce Nonce to validate (may contain an encoded space)
    * @dev A valid nonce must be above the last one used
    *   with a maximum delta of 100
    */
-  function _validateNonce(uint256 _nonce) private {
+  function _validateNonce(uint256 _rawNonce) private {
     // Retrieve current nonce for this wallet
-    uint256 current_nonce = nonce; // Lowest valid nonce for signer
+    (uint256 space, uint256 providedNonce) = _decodeNonce(_rawNonce);
+    uint256 currentNonce = readNonce(space);
 
     // Verify if nonce is valid
     require(
-      (_nonce >= current_nonce) && (_nonce < (current_nonce + 100)),
+      providedNonce == currentNonce,
       "MainModule#_auth: INVALID_NONCE"
     );
 
     // Update signature nonce
-    nonce = _nonce + 1;
-    emit NonceChange(_nonce + 1);
+    providedNonce++;
+    _writeNonce(space, providedNonce);
```

Comment:
I would put 122 and 123 in same line so it's a bit more explicit that we store the nonce + 1, maybe 
`_writeNonce(space, providedNonce + 1);`

Reply:
I wanted to avoid performing the `add` operation twice, but I think that it can be a little bit more explicit by using a `newNonce` variable

Reply:
Right, because of the event. I think the clarity from having new nonce is nice, the cost difference is probably negligible.